src/extra/treemacs-all-the-icons.el

```diff
@@ -252,22 +224,27 @@
                                        "tr3" "oxps" "xps")
                           :fallback 'same-as-icon)
     (treemacs-create-icon :icon (format "  %s%s" (all-the-icons-octicon "file-text" :v-adjust 0 :face 'treemacs-all-the-icons-file-face) treemacs-all-the-icons-tab)
-                          :extensions ("md" "markdown" "rst" "log" "txt"
-                                       "CONTRIBUTE" "LICENSE" "README" "CHANGELOG")
+                          :extensions ("markdown" "rst" "CONTRIBUTE" "LICENSE" "README" "CHANGELOG")
                           :fallback 'same-as-icon)
     (treemacs-create-icon :icon (format "  %s%s" (all-the-icons-octicon "file-binary" :v-adjust 0 :face 'treemacs-all-the-icons-file-face) treemacs-all-the-icons-tab)
-                          :extensions ("class" "exe" "dll" "obj" "so" "o" "out" "elc" "pyc")
-                          :fallback 'same-as-icon)
-    (treemacs-create-icon :icon (format "  %s%s" (all-the-icons-octicon "file-pdf" :v-adjust 0 :face 'all-the-icons-orange) treemacs-all-the-icons-tab)
-                          :extensions ("pdf")
+                          :extensions ("class" "exe" "obj" "so" "o" "out" "pyc")
                           :fallback 'same-as-icon)
     (treemacs-create-icon :icon (format "  %s%s" (all-the-icons-octicon "file-zip" :v-adjust 0 :face 'treemacs-all-the-icons-file-face) treemacs-all-the-icons-tab)
-                          :extensions ("zip" "7z" "tar" "gz" "rar" "tgz")
+                          :extensions ("tar" "rar" "tgz")
                           :fallback 'same-as-icon)
 
     (treemacs-create-icon :icon (format "  %s%s" (all-the-icons-octicon "file-text" :v-adjust 0 :face 'treemacs-all-the-icons-file-face) treemacs-all-the-icons-tab)
                           :extensions (fallback)
-                          :fallback 'same-as-icon)))
+                          :fallback 'same-as-icon))
+  )
+
+
+(treemacs-create-theme "all-the-icons"
+  :config
+  (eval (cons 'progn (append
```

Comment:
`eval` is a code smell, and a strong hint that you need to rethink your approach. In this case I assume it was necessary because `treemacs-create-icon` makes it very difficult to programmatically get the extensions correct. It's a macro that was only ever meant for declarative use, and working around its limitations is admittedly very tricky. 

It's also been annoying me for a long time now, so give me a bit, I'll try to finally have a useable workaround over the weekend.